datalad/interface/common_cfg.py

```diff
@@ -28,9 +29,185 @@
 )
 from datalad.utils import on_windows
 
+lgr = logging.getLogger('datalad.interface.common_cfg')
 dirs = AppDirs("datalad", "datalad.org")
 
 
+class _NotGiven():
+    pass
+
+
+def register_config(
```

Comment:
I think, this lacks a conflict prevention. While it's okay to override existing configs from core (such as changing a default value), name conflict with other extensions is likely unintentional. May be override should require a `force=True` to make clear that it is intended.

Reply:
I cannot see this having a net-positive effect. The person to set `force=True` is a developer of an extension. They, as any other person, will not see the big picture of all extensions and future versions of core. So not setting `force=True` has the "benefit" of informing *users* at some point that they cannot use an extension (anymore). Their only cause of action is to inform developers of the affected packages and hope for a timely release of an update. As an extension developer, i can prevent this inconvenience by always setting `force=True` (which is what I implemented), and which is not change from the status quo. IMHO it is an illusion that 'conflict prevention' is done in the scope of this function. If there ever is a conflict, the resolution will need to come with a (large-scale) migration of existing configuration users have lying around on any system and clone, or in any dataset branch.

I believe the only approach here to avoid conflicts is proper monitoring of developers by developers, and good changelogs.

That being said, if anyone has a good idea how to approach this in a meaningful way, please propose a solution in a dedicated PR.

Reply:
Good points. Agree with addressing elsewhere if and when there's a better idea.